### ISSUE TITLE
CopyQ not copying all of the time in Kali Linux

### ISSUE THREAD

**Issue by robertstrom** [hluk/CopyQ#1975]
Before creating a new issue, see list of [known
issues](https://copyq.readthedocs.io/en/latest/known-issues.html).

**Describe the bug**

I have an instance of CopyQ running on my Kali Linux and it will mysteriously just stop copying selected text in both the  terminals (I am using both the stock QT Terminal and the Terminator terminal) and in various GUI applications (one example is the Geany IDE / text editor).

The log file included was specifically only running while I was moving from terminal to terminal and to the Geany application and selecting text within each new window and copying the text. In almost every situation the text did not appear in the CopyQ clipboard  tab as expected. I could, however, move the cursor to an application outside of the Kali virtual machine (Joplin) and paste the text that I just copied from the Geany application in the Kali VM (yet the text appears to be nowhere in CopyQ itself).

I have found that if CopyQ will not copy text that I want copied from a terminal window I can place the text in quotes and place the echo command in front of the quoted text and then pipe the echoed text to the **xclip** program and the text will then appear in the CopyQ clipboard tab. Example:

```
echo "find ./ -type d \( -path ./QNAPMyDocs -o -path ./.vscode \) -prune -o -iname '*.py' -print" | xclip
```

**To Reproduce**

Described above

**Expected behavior**

Copy all text copied from terminals and GUI program to the CopyQ clipboard and save them there until deleted or they roll off due to the clipboard tab exceeding the maximum number of clips configured.

**Screenshots**

Really no screenshots to be taken of this. I am including the DEBUG logs.

**NOTE:  The FAQ page that has the DEBUG log instructions has a typo**
https://copyq.readthedocs.io/en/latest/faq.html#how-to-enable-logging

These instructions have an incorrect slash (it is a backslash and should be a forward slash) in the third line:
```
copyq exit
export COPYQ_LOG_LEVEL='DEBUG'
export COPYQ_LOG_FILE="$HOME\copyq.log"
echo "Logs will be written to $COPYQ_LOG_FILE"
copyq
```

The commands should be (with the third line having a forward slash for Mac and Linux):
```
copyq exit
export COPYQ_LOG_LEVEL='DEBUG'
export COPYQ_LOG_FILE="$HOME/copyq.log"
echo "Logs will be written to $COPYQ_LOG_FILE"
copyq
```

**Version, OS and Environment**
(Get details from `copyq version` command if possible.)
 - Application Version [e.g. 3.7.2]
 - OS [e.g. Windows]
 - Desktop environment, window manager (if applicable)

CopyQ 6.1.0

```
┌──(rstrom㉿kali-2022)-[~/exploits]
└─$ cat /etc/os-release                                                                            130 ⨯
PRETTY_NAME="Kali GNU/Linux Rolling"
NAME="Kali GNU/Linux"
ID=kali
VERSION="2022.1"
VERSION_ID="2022.1"
VERSION_CODENAME="kali-rolling"
ID_LIKE=debian
ANSI_COLOR="1;31"
HOME_URL="https://www.kali.org/"
SUPPORT_URL="https://forums.kali.org/"
BUG_REPORT_URL="https://bugs.kali.org/"
```

The Kali Linux is using the stock window manager (XFCE I believe)

**Additional context**

FYI - the last clip from the Kali VM did copy and make it to CopyQ (it's in the DEBUG logs if that helps to define what worked and what did not)

[copyq.log](https://github.com/hluk/CopyQ/files/8598101/copyq.log)

**Comment by hluk** [comment]
> NOTE: The FAQ page that has the DEBUG log instructions has a typo

Thanks, fixed.

**Comment by hluk** [comment]
> I could, however, move the cursor to an application outside of the Kali virtual machine (Joplin) and paste the text that I just copied from the Geany application in the Kali VM (yet the text appears to be nowhere in CopyQ itself).

This sounds like it could be an issue with virtual machine software/integration. There has been a few in the past.

**Comment by robertstrom** [comment]
I am using VMware Workstation 16 on a Windows 11 host, Kali Linux guest.

So what to do about this?

Thanks,

Robert

**Comment by hluk** [comment]
Not sure how to fix this. You can try using flatpak version of CopyQ (see https://copyq.readthedocs.io/en/latest/installation.html).

**Comment by robertstrom** [comment]
K, flatpak appears to be supported on Kali. I will try

What is the difference between running the .deb version and the flatpak version?

Thanks!

**Comment by robertstrom** [comment]
I have uninstalled the .deb version of CopyQ and installed flatpak and then installed the flatpak version of CopyQ. I have imported a backup which did import my data

Initial virgin install of CopyQ flatpak
![image](https://user-images.githubusercontent.com/3240205/166615294-b77318d4-c269-4107-9a82-997d322dc75b.png)

CopyQ import
![image](https://user-images.githubusercontent.com/3240205/166615395-4d20d557-ea0c-48b9-9c1f-7ebdb5eacfaa.png)
![image](https://user-images.githubusercontent.com/3240205/166615454-9cb61572-afbb-49ee-81b0-44066147a19d.png)

CopyQ import done - note **clipboard** and **clipboard (1)**. I am aware that **clipboard (1)** is the contents of my backup **clipboard** tab. So there are 4 items in the new **clipboard** tab and a **?** quantity for all other tabs (CopyQ does not know until you click on each tab
![image](https://user-images.githubusercontent.com/3240205/166615508-5a9e4b60-35ed-48d6-b330-de1c7c9634c8.png)

Click on **clipboard (1)** tab and we see 6000 items
![image](https://user-images.githubusercontent.com/3240205/166615706-d91f80ff-b54b-4bd4-8df7-aa0d32bec35a.png)

Note 350 items in the **url** tab
![image](https://user-images.githubusercontent.com/3240205/166615782-6f531f9d-fb24-45c3-8357-c9e9d7ef3beb.png)

Removing the **clipboard** tab so that the current **clipboard (1)** tab can become the main **clipboard** tab
![image](https://user-images.githubusercontent.com/3240205/166615870-c0e895c8-4da4-4092-bc7b-cf8896f398da.png)

**clipboard** tab deleted
![image](https://user-images.githubusercontent.com/3240205/166615909-da23a940-c64f-44dd-afe0-ce55795e1743.png)

At this point I am usually able to rename the **clipboard (1)** tab to **clipboard** but t is not allowed at this time so I shut down CopyQ (File / Exit)
![image](https://user-images.githubusercontent.com/3240205/166615998-d59cb282-dc82-428e-a112-1c92be1577af.png)

I renamed the **clipboard (1)** tab to **clipboard** in the copyq.conf file and now the data is gone (I had, before starting all this documentation on this this evening, been able to rename the tab and had the same results. I have done this countless times with the .deb version of the app and it works without issue / as expected. Also note that the contents of the **url** tab is now empty. Other tabs appear to still have their data.
![image](https://user-images.githubusercontent.com/3240205/166616580-0249b403-fa0d-4184-ade2-306f67a5608b.png)
![image](https://user-images.githubusercontent.com/3240205/166617361-7bf0a38c-3cdd-47cd-9994-9b1cbc86e920.png)

Rename the tab back to **clipboard (1)** in the copyq.conf file and the data is back in that tab but the **url** tab data is still missing
![image](https://user-images.githubusercontent.com/3240205/166616897-50025aa6-8293-4e6c-9ef4-5440284ddfb9.png)
![image](https://user-images.githubusercontent.com/3240205/166616921-543f1fb4-992f-4b09-8f73-407ba9d12400.png)

Also note that the import did not appear to import my settings properly. Example - the tab labeled **clipboard (1)** should have a Maximum number of items set to 6000 and it is set to the default
![image](https://user-images.githubusercontent.com/3240205/166617142-8e92cdae-b62c-40d7-81b7-cb33b1893eb3.png)

Also note that while it appears that my list of commands appears to have been imported, including the **Backup On Exit**, the backup is not doing anything. I have shut CopyQ down many times since starting this and there are no new backups. The newest backup is about 5 hours ago.
![image](https://user-images.githubusercontent.com/3240205/166617273-bd729e73-5fc8-4261-bc36-0e03e0e78c04.png)
![image](https://user-images.githubusercontent.com/3240205/166617318-a6e4def7-ee05-4210-8550-6af544de1864.png)

I deleted all the data and set the logging to DEBUG (you need to document how to do this with a flatpak install if it is not already) and imported the backup and then renamed the new, empty **clipboard** tab to **clipboard-old** and then I was able to rename the **clipboard (1)** tab to **clipboard** and all my data appears to be there now (in both the **clipboard** and the **url** tabs).
![image](https://user-images.githubusercontent.com/3240205/166618062-88dcbad3-ff1e-4490-9137-2445276a8d55.png)

Now to close the program and reopen it and see what happens. Both the **clipboard** and the **url** tabs are now empty
![image](https://user-images.githubusercontent.com/3240205/166618148-096e5ecc-9474-40bf-b6df-8af54705ff0f.png)

The DEBUG logs that I configured do not appear to exist
![image](https://user-images.githubusercontent.com/3240205/166618238-2f72d892-4102-4da2-b2c2-641745d0fa5f.png)

A find for **.log** files inthe flatpak directory
![image](https://user-images.githubusercontent.com/3240205/166618437-3fd94c2e-679e-4327-b758-99e874704dbf.png)

A find for the named DEBUG log file in any of my home directory
![image](https://user-images.githubusercontent.com/3240205/166618522-fa8a9bb9-67bc-4a57-9df1-ade9c72844ef.png)

So far not really liking flatpak very much ...

I'm fine with trying the flatpak version to see if it will solve my issues, but it has created a number of new challenges.

Given this information (I would have happily supplied the DEBUG logs but ...) what do I do next?

HELP Please!

Thanks!

Robert

**Comment by robertstrom** [comment]
Just as a test I uninstalled **xclip** and tested to see if that made any difference (this is in the deb install environment) and it did not. As a precaution I rebooted and tested again. Still no difference.

I have reinstalled **xclip**

**Comment by robertstrom** [comment]
Touching base. Any questions, comments, suggestions, updates to try? Anything needed from me here on this issue?

Thanks

**Comment by hluk** [comment]
I haven't been able to respond to issues in a while and there is too much to unpack here.

I tested config files you provided and they seem to work (at least on Fedora 36).

From the last screenshot it seems that CopyQ detected the copy operation from the terminal, it just did not add it to the clipboard tab. Is that correct?

> If I have CopyQ up and running and have Clipman up and running at the same time and select text in the terminal, text in the web browser, text in Geany all of it is copied to the clipboard history and stored

This is very strange. There is nothing that would require CopyQ to run additional clipboard manager under X11.

**Comment by robertstrom** [comment]
> From the last screenshot it seems that CopyQ detected the copy operation from the terminal, it just did not add it to the clipboard tab. Is that correct?

Yes, that is correct. I saw this exact behavior again yesterday.

The configuration that I have having the **huge** issue with is a Windows 11 host running Kali Linux 2022.2 using XFCE desktop inside VMware Workstation 16.2.3. I am taking the OSCP PEN 200 course and have been using this setup very heavily and the documentation of what I am doing relies on tons of copying and pasting both within the VM itself and between the VM and the host OS.

> This is very strange. There is nothing that would require CopyQ to run additional clipboard manager under X11.

Agreed but right now it is my fallback for when CopyQ does not capture the snippet that I need.

I love CopyQ but this is very frustrating.

Is your Fedora a bare metal system or a VM?

I have a couple of other Kali VM's that I have (two are using KDE). They are not my day to day Kali instances but I will try to determine if they are also experiencing the same issue. I also have two bare metal systems running Pop!_OS (based on current Ubuntu 22.04 LTS) that I use. Both have CopyQ installed but again, the copy and paste withing the Kali instance is very heavy. Clips of code from the terminal and lots of screenshots. None of my other setups have this amount of copying and pasting.

**Comment by robertstrom** [comment]
copyq -s test appears to work / that's GREAT .. .except ... I have removed
every single F6 command from my instance and then tried copying and nothing
is getting captured.

When I delete all of the commands, what appears to be a predefined set of
commands is automatically populated. So at this time those are the only
commands that are currently in the F6 command list.

[image: image.png]

So, it seems like you are on the right track. Just need to figure out how
to get copying again with all of my stored clips and where to go from there
to get me some commands back.

Thanks!

**Comment by robertstrom** [comment]
Clarification on the previous reply. copyq -s test works great ... and then
I tried to run with my instance and remove all of the commands and that did
not work.

But ...

Looks like another step forward in the right direction.

I launched the copyq -s test and then imported my backup without the
commands and that did not work.

I deleted the *copyq-test* instance and then launched again and imported a
backup without the commands and without the configuration and that appears
to be successful in as much as I have tested in a few minutes.

One issue I see with this is that the default maximum for the *clipboard *tab
(I believe any tab) is 200, so I lose the rest of my 6k clips from my
current clipboard tab (I also had 300 in a URL's tab but I could live
without that).

Think I just got that figured out. I exported the *clipboard* tab from the
old instance and then imported it to the new instance and then deleted the
*clipboard* tab and renamed the *clipboard (1)* tab to *clipboard* and got
past that point.

I may be good to go now. I am going to move the *copyq-test* instance back
to the standard *copyq* instance and proceed from there. I have all my old
backups so I will be able to revert if needed.

I will start adding back some of the commands and see where that takes me.

Thanks for thinking of that one little thing that led me down this
troubleshooting path. Not sure about you, but it seems like something in
the configuration was corrupted.

CopyQ ... and its developers, do rock! ;-)

Thanks,

Robert

**Comment by Chealer** [comment]
Thanks for reporting @robertstrom, but please avoid commenting via email. Your first comment from 2022-06-13 is broken.

Does this persist with CopyQ 13 or ulterior? If so, please edit the description and actually fill the _To Reproduce_ section following [this guidance](https://blog.michael.gr/post/2014-11-bug-reporting-checklist/#10-list-the-steps-to-reproduce-the-problem) (not just with "Described above").
